src/outputs/ascent.cpp

```diff
@@ -0,0 +1,203 @@
+    auto jb_int = bounds.GetBoundsJ(IndexDomain::interior);
+    auto kb_int = bounds.GetBoundsK(IndexDomain::interior);
+
+    auto &coords = pmb->coords;
+    Real dx1 = coords.CellWidth<X1DIR>(ib.s, jb.s, kb.s);
+    Real dx2 = coords.CellWidth<X2DIR>(ib.s, jb.s, kb.s);
+    Real dx3 = coords.CellWidth<X3DIR>(ib.s, jb.s, kb.s);
+    std::array<Real, 3> corner = coords.GetXmin();
+
+    // create the coordinate set
+    mesh["coordsets/coords/type"] = "uniform";
+
+    mesh["coordsets/coords/dims/i"] = nx + 1;
+    mesh["coordsets/coords/dims/j"] = ny + 1;
+    if (nz > 1) {
+      mesh["coordsets/coords/dims/k"] = nz + 1;
+    }
+
+    // add origin and spacing to the coordset (optional)
+    mesh["coordsets/coords/origin/x"] = corner[0];
+    mesh["coordsets/coords/origin/y"] = corner[1];
+    if (nz > 1) {
+      mesh["coordsets/coords/origin/z"] = corner[2];
+    }
+
+    mesh["coordsets/coords/spacing/dx"] = dx1;
+    mesh["coordsets/coords/spacing/dy"] = dx2;
+    if (nz > 1) {
+      mesh["coordsets/coords/spacing/dz"] = dx3;
+    }
+
+    // add the topology
+    mesh["topologies/topo/type"] = "uniform";
+    mesh["topologies/topo/coordset"] = "coords";
+
+    // indicate ghost zones with ascent_ghosts set to 1
+    Node &n_field = mesh["fields/ascent_ghosts"];
+    n_field["association"] = "element";
+    n_field["topology"] = "topo";
+    n_field["values"].set(conduit::DataType::int32(ncells)); // NOLINT
+    conduit::int32_array vals_array = n_field["values"].value();
+
+    int idx = 0;
+    for (int k = kb.s; k <= kb.e; k++) {
+      for (int j = jb.s; j <= jb.e; j++) {
+        for (int i = ib.s; i <= ib.e; i++) {
+          if ((i < ib_int.s) || (ib_int.e < i) || (j < jb_int.s) || (jb_int.e < j) ||
+              ((nz > 1) && ((k < kb_int.s) || (kb_int.e < k)))) {
+            vals_array[idx] = 1;
+          } else {
+            vals_array[idx] = 0;
+          }
+          idx++;
+        }
+      }
+    }
+
+    // create a field for each component of each variable pack
+    auto &mbd = pmb->meshblock_data.Get();
+
+    for (const auto &var : mbd->GetCellVariableVector()) {
+      PARTHENON_REQUIRE(var->GetDim(6) * var->GetDim(5) == 1,
+                        "Only scalar and vector fields are currently exported in Ascent. "
+                        "Please open an issue on GitHub if you have a use case.");
+
+      const std::string packname = var->label();
+      auto const &labels = var->metadata().getComponentLabels();
+
+      for (int icomp = 0; icomp < labels.size(); ++icomp) {
+        auto const data = Kokkos::subview(var->data, 0, 0, icomp, Kokkos::ALL(),
+                                          Kokkos::ALL(), Kokkos::ALL());
+        const std::string varname = packname + ":" + labels.at(icomp);
+        mesh["fields/" + varname + "/association"] = "element";
+        mesh["fields/" + varname + "/topology"] = "topo";
+        mesh["fields/" + varname + "/values"].set_external(data.data(), ncells);
+      }
```

Comment:
So if I understand this correctly, it also means that for vector fields, we need to assign labels?

Reply:
Good point. I'll replicate/reuse the logic we use in the hdf5 outputs, i.e.,
```
    component_labels = {};
    if (vlen == 1 || is_vector) {
      component_labels = component_labels_.size() > 0 ? component_labels_
                                                      : std::vector<std::string>({label});
    } else if (component_labels_.size() == vlen) {
      component_labels = component_labels_;
    } else {
      for (int i = 0; i < vlen; i++) {
        component_labels.push_back(label + "_" + std::to_string(i));
      }
```

Reply:
:+1:

Reply:
I now centralized the `VarInfo` struct to `outputs.#pp` so that it could be reused here for the labels.